packages/fxa-settings/src/pages/InlineTotpSetup/index.tsx

```diff
@@ -0,0 +1,397 @@
+import React, { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import classNames from 'classnames';
+import { MozServices } from '../../lib/types';
+import { FtlMsg } from 'fxa-react/lib/utils';
+import { useFtlMsgResolver } from '../../models';
+import { logViewEvent, logPageViewEvent } from '../../lib/metrics';
+import { ReactComponent as TwoFactorImg } from '../Signin/SigninTotpCode/graphic_two_factor_auth.svg';
+import InputText from '../../components/InputText';
+import LinkExternal from 'fxa-react/components/LinkExternal';
+
+type InlineTotpSetupProps = {
+  code: string;
+  serviceName?: MozServices;
+};
+
+export const InlineTotpSetup = ({
+  code,
+  serviceName,
+}: InlineTotpSetupProps) => {
+  logPageViewEvent('inline-totp-setup', { entrypoint_variation: 'react' });
+
+  /*
+   * TODO:
+   *  - Write functionality to get TOTP token from account
+   *  - get secret and QR code image src from totp token
+   *  - add in success and error messages (with Banner component)
+   *  - add in actions for `Cancel` `Ready` etc
+   *  - fetch code here if that's preferable
+   */
+
+  const ftlMsgResolver = useFtlMsgResolver();
+  const localizedSecurityCodePlaceholder = ftlMsgResolver.getMsg(
+    'inline-totp-setup-security-code-placeholder',
+    'Security code'
+  );
+  const localizedQRCodeAltText = ftlMsgResolver.getMsg(
+    'inline-totp-setup-qr-code-alt-text',
+    `Use the code ${code} to set up two-step authentication in supported applications.`,
+    { code }
+  );
+  const localizedTwoFactorImgAriaLabel = ftlMsgResolver.getMsg(
+    'inline-totp-setup-image-label',
+    'A device with a hidden 6-digit code.'
+  );
+  const [secret, setSecret] = useState<string>();
+  const [qrCodeSrc, setQRCodeSrc] = useState<string>();
+  const [showIntro, setShowIntro] = useState<boolean>(true);
+  const [totpCodeValue, setTotpCodeValue] = useState<string>('');
+  const [showQR, setShowQR] = useState<boolean>(true);
+  const [totpErrorMessage, setTotpErrorMessage] = useState<string>('');
+
+  const [isFocused, setIsFocused] = useState(false);
```

Comment:
nit: Should this be `useState<boolean>(false)` for consistency w/ L46-51 above?
```suggestion
  const [isFocused, setIsFocused] = useState<boolean>(false);
```

Reply:
It should! Thank you!